cardano-api/src/Cardano/Api/SpecialByron.hs

```diff
@@ -55,11 +55,11 @@ instance HasTypeProxy ByronUpdateProposal where
 
 instance SerialiseAsRawBytes ByronUpdateProposal where
   serialiseToRawBytes (ByronUpdateProposal proposal) = annotation proposal
-  deserialiseFromRawBytes AsByronUpdateProposal bs =
+  eitherDeserialiseFromRawBytes AsByronUpdateProposal bs =
     let lBs = LB.fromStrict bs
     in case Binary.decodeFull lBs of
-        Left _deserFail -> Nothing
-        Right proposal -> Just (ByronUpdateProposal proposal')
+        Left e -> Left $ "Unable to deserialise ByronUpdateProposal: " <> show e
```

Comment:
In order to (more easily) maintain consistency in the errors what do you think about the following:
```
...
 eitherDeserialiseFromRawBytes
    :: AsType a -> ByteString -> Either (SerialiseAsRawBytesError (AsType a)) a

data SerialiseAsRawBytesError a = SerialiseAsRawBytesError a

instance Error (SerialiseAsRawBytesError a) where
  displayError (SerialiseAsRawBytesError a) = "Unable to deserialise " <> show a
```
Example usage: 
```
...
    eitherDeserialiseFromRawBytes AsScriptHash bs =
      maybeToRight (SerialiseAsRawBytesError AsScriptHash) $
        ScriptHash . Shelley.ScriptHash <$> Crypto.hashFromBytes bs
...
```

Reply:
You could potentially also include the original `ByteString`

Reply:
If we did this we would only have one error value per `AsType`, so we couldn't have more than one kind of serialisation error per type.  If deserialisation could fail in more than one way depending on what is in the byte string we would be unable to communicate to the user that distinction.

Reply:
For the moment, I created a new type `SerialiseAsRawBytesError` which is just a wrapper around `String` whilst waiting for further feedback.